### DISCUSSION TITLE
Does juicefs csi driver support tls on redis?

### DISCUSSION THREAD

**Discussion by FrostyLeaf** [juicedata/juicefs#1292] [Q&A] [answered]
Recently I'm trying to set up an AWS ElastiCache for Redis to work with JuiceFS. In the AWS ElasticCache service, TLS is needed by both Redis AUTH and RBAC. I tried Redis AUTH (which means I had to enable TLS), always got an error like this:

```shell
E0113 08:59:26.221671       1 driver.go:60] GRPC error: rpc error: code = Internal desc = Could not mount juicefs: rpc error: code = Internal desc = Could not format juicefs: exit status 1
I0113 08:59:45.317030       1 juicefs.go:170] JfsMount: format output is '2022/01/13 08:55:35.527890 juicefs[306] <INFO>: Meta address: redis://@<redis-master>:6379
2022/01/13 08:57:06.685676 juicefs[306] <WARNING>: parse info: i/o timeout
2022/01/13 08:57:06.686060 juicefs[306] <INFO>: Data use s3://<bucket>/jfs-default-storage-secret/
2022/01/13 08:59:45.314206 juicefs[306] <FATAL>: format: EOF
```

It looks like the `i/o timeout` message implies JuiceFS never use TLS to connect to Redis. No luck finding anything related from the docs.

EC2 instance or any pod in the same EKS cluster with a TLS compiled Redis client works fine with the TLS enabled ElastiCache Redis, and JuiceFS also works fine with a self-hosted Redis with AUTH enabled, BTW.

So, as the title asked, does the JuiceFS CSI driver support TLS on Redis? did I miss something from the docs?

**Comment by davies** [comment, marked as answer]
JuiceFS support TLS on Redis with the `rediss://HOST:PORT`, please use the `rediss://:PASSWORD@HOST:PORT` as the meta-url.

**Reply by FrostyLeaf** [reply]
Get it, thanks!
